packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -16,6 +16,8 @@ export default function copyEmptyArrayProps(oldObj, newObj) {
       if (foundKey === -1) {
         newObj[key] = [] // eslint-disable-line no-param-reassign
       }
+    } else if (typeof value === 'object' && oldObj[key] && newObj[key]) {
```

Comment:
I don't understand this condition. If  `oldObj[key] && newObj[key]` is true, what needs to be fixed then? In that case both objects already have a defined value and there is no need to fix them. Or am I wrong here?

Reply:
You right, i should add more descriptive comment

Its recursion, so if we have obj like this one 
`{ 
  prop1: []
  obj: {
    prop2: []
  }
}`

with the above condition, then we copy both prop1 and prop2, without this condition then we will just copy prop1 which will cause the issue `TypeError: Cannot read property '0' of undefined`